utilities/test_suite/HIP/Tensor_hip.cpp

```diff
@@ -0,0 +1,1877 @@
+        {
+            hipMemcpy(output, d_output, outputBufferSize, hipMemcpyDeviceToHost);
+            memcpy(outputu8, output, outputBufferSize);
+        }
+        else if ((inputBitDepth == 1) || (inputBitDepth == 3))
+        {
+            hipMemcpy(output, d_output, outputBufferSize, hipMemcpyDeviceToHost);
+            Rpp8u *outputTemp;
+            outputTemp = outputu8 + dstDescPtr->offsetInBytes;
+            half *outputf16Temp;
+            outputf16Temp = (half *)((Rpp8u *)output + dstDescPtr->offsetInBytes);
+            for (int i = 0; i < oBufferSize; i++)
+            {
+                *outputTemp = (Rpp8u)RPPPIXELCHECK((float)*outputf16Temp * invConversionFactor);
+                outputf16Temp++;
+                outputTemp++;
+            }
+        }
+        else if ((inputBitDepth == 2) || (inputBitDepth == 4))
+        {
+            hipMemcpy(output, d_output, outputBufferSize, hipMemcpyDeviceToHost);
+            Rpp8u *outputTemp;
+            outputTemp = outputu8 + dstDescPtr->offsetInBytes;
+            Rpp32f *outputf32Temp;
+            outputf32Temp = (Rpp32f *)((Rpp8u *)output + dstDescPtr->offsetInBytes);
+            for (int i = 0; i < oBufferSize; i++)
+            {
+                *outputTemp = (Rpp8u)RPPPIXELCHECK(*outputf32Temp * invConversionFactor);
+                outputf32Temp++;
+                outputTemp++;
+            }
+        }
+        else if ((inputBitDepth == 5) || (inputBitDepth == 6))
+        {
+            hipMemcpy(output, d_output, outputBufferSize, hipMemcpyDeviceToHost);
+            Rpp8u *outputTemp = outputu8 + dstDescPtr->offsetInBytes;
+            Rpp8s *outputi8Temp = (Rpp8s *)output + dstDescPtr->offsetInBytes;
+            for (int i = 0; i < oBufferSize; i++)
+            {
+                *outputTemp = (Rpp8u) RPPPIXELCHECK(((Rpp32s) *outputi8Temp) + 128);
+                outputi8Temp++;
+                outputTemp++;
+            }
+        }
+
+        // Calculate exact dstROI in XYWH format for OpenCV dump
+        if (roiTypeSrc == RpptRoiType::LTRB)
+        {
+            for (int i = 0; i < dstDescPtr->n; i++)
+            {
+                int ltX = roiTensorPtrSrc[i].ltrbROI.lt.x;
+                int ltY = roiTensorPtrSrc[i].ltrbROI.lt.y;
+                int rbX = roiTensorPtrSrc[i].ltrbROI.rb.x;
+                int rbY = roiTensorPtrSrc[i].ltrbROI.rb.y;
+
+                roiTensorPtrSrc[i].xywhROI.xy.x = ltX;
+                roiTensorPtrSrc[i].xywhROI.xy.y = ltY;
+                roiTensorPtrSrc[i].xywhROI.roiWidth = rbX - ltX + 1;
+                roiTensorPtrSrc[i].xywhROI.roiHeight = rbY - ltY + 1;
+            }
+        }
+
+        if(inputBitDepth == 0 && (srcDescPtr->layout == dstDescPtr->layout))
+            compare_output<Rpp8u>(outputu8, func, testCaseName, dstDescPtr, "HIP");
+
+        RpptROI roiDefault;
+        RpptROIPtr roiPtrDefault;
+        roiPtrDefault = &roiDefault;
+        roiPtrDefault->xywhROI.xy.x = 0;
```

Comment:
Check for possible struct initialization changes like this throughout the file:
```
RpptROI roiDefault = {0, 0, dstDescPtr->w, dstDescPtr->h};
RpptROIPtr roiPtrDefault = &roiDefault;
```

Reply:
Done